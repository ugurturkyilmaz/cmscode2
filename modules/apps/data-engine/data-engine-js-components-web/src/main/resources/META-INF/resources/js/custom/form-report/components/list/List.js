/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ClayButton from '@clayui/button';
import moment from 'moment/min/moment-with-locales';
import React, {useContext} from 'react';

import {removeEmptyValues} from '../../utils/data';
import Color from '../color/Color';
import {SidebarContext} from '../sidebar/SidebarContext';

export default function List({data, field, summary, totalEntries, type}) {
	const {portletNamespace, toggleSidebar} = useContext(SidebarContext);

	const formatDate = (field, isDateTime) => {
		const locale = themeDisplay.getLanguageId().split('_', 1).join('');
		const date = moment(field).locale(locale).format('L');

		if (!isDateTime) {
			return date;
		}

		const time = moment(field).locale(locale).format('LT');

		return `${date} ${time}`;
	};

	const checkType = (field, type) => {
		switch (type) {
			case 'color':
				return <Color hexColor={field} />;
			case 'date':
				return formatDate(field);
			case 'date_time':
				return formatDate(field, true);
			default:
				return field;
		}
	};

	data = removeEmptyValues(data);

	return (
		<div className="field-list">
			<ul className="entries-list">
				{Array.isArray(data) &&
					data.map((field, index) => (
						<li key={index}>{checkType(field, type)}</li>
					))}

				{data.length === 5 && totalEntries > 5 ? (
					<li id={`${portletNamespace}-see-more`} key="see-more">
						<ClayButton
							displayType="link"
							onClick={() =>
								toggleSidebar(
									field,
									summary,
									totalEntries,
									type
								)
							}
						>
							{Liferay.Language.get('see-all-entries')}
						</ClayButton>
					</li>
				) : null}
			</ul>
		</div>
	);
}
