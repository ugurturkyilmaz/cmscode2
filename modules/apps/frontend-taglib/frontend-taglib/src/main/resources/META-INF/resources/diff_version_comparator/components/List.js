/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ClayAlert from '@clayui/alert';
import ClayList from '@clayui/list';
import React from 'react';

export default function List({onChange, selected, versions}) {
	return (
		<>
			{versions.length ? (
				<ClayList>
					{versions.map((version) => {
						return (
							<ClayList.Item
								active={
									selected &&
									selected.version === version.version
								}
								className="version-item"
								data-version={version.version}
								flex
								key={version.version}
								onClick={onChange}
							>
								<ClayList.ItemField>
									<div className="sticker sticker-secondary sticker-user-icon user-icon-color-0">
										<span>{version.userInitials}</span>
									</div>
								</ClayList.ItemField>

								<ClayList.ItemField expand>
									<ClayList.ItemText>
										{version.displayDate}
									</ClayList.ItemText>

									<ClayList.ItemTitle>
										{version.label}
									</ClayList.ItemTitle>

									<ClayList.ItemText>
										{version.userName}
									</ClayList.ItemText>
								</ClayList.ItemField>
							</ClayList.Item>
						);
					})}
				</ClayList>
			) : null}

			{!versions.length && (
				<ClayAlert displayType="info">
					{Liferay.Language.get('there-are-no-results')}
				</ClayAlert>
			)}
		</>
	);
}
