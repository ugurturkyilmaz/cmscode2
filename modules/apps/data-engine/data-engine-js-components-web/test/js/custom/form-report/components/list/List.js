/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {render} from '@testing-library/react';
import React from 'react';

import List from '../../../../../../src/main/resources/META-INF/resources/js/custom/form-report/components/list/List';

const props = {
	data: ['label1', 'label2', 'label3', 'label4', 'label5'],
	onClick: () => {},
	summary: {},
	totalEntries: 5,
};

describe('List', () => {
	it('renders', () => {
		const {asFragment, container} = render(<List {...props} />);

		const li = container.querySelectorAll('li');

		expect(li.length).toBe(5);

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders color list', () => {
		const colorProps = {
			...props,
			data: ['7F26FF', '2BA676', 'CBCBCB', 'FF21A0', 'FF0D0D'],
			type: 'color',
		};

		const {container} = render(<List {...colorProps} />);

		const colorTextList = container.querySelectorAll('.color-text');

		expect(colorTextList.length).toBe(5);

		colorTextList.forEach((colorText, index) =>
			expect(colorText.textContent).toBe(colorProps.data[index])
		);
	});

	it('shows a button to see all entries when there are more than 5 entries', () => {
		const {container} = render(<List {...props} totalEntries={6} />);

		expect(container.querySelector('button').innerHTML).toBe(
			'see-all-entries'
		);
	});

	it('renders dates in the english US format', () => {
		const originalThemeDisplay = themeDisplay;
		themeDisplay = {getLanguageId: () => 'en_US'};

		const {getByText} = render(
			<List {...props} data={['2020-12-20']} type="date" />
		);

		expect(getByText('12/20/2020')).toBeTruthy();

		themeDisplay = originalThemeDisplay;
	});

	it('renders dates in the portuguese BR format', () => {
		const originalThemeDisplay = themeDisplay;
		themeDisplay = {getLanguageId: () => 'pt_BR'};

		const {getByText} = render(
			<List {...props} data={['2020-12-20']} type="date" />
		);

		expect(getByText('20/12/2020')).toBeTruthy();

		themeDisplay = originalThemeDisplay;
	});

	it('renders dates and time in the english US format', () => {
		const originalThemeDisplay = themeDisplay;
		themeDisplay = {getLanguageId: () => 'en_US'};

		const {getByText} = render(
			<List {...props} data={['2020-12-20 14:48']} type="date_time" />
		);

		expect(getByText('12/20/2020 2:48 PM')).toBeTruthy();

		themeDisplay = originalThemeDisplay;
	});

	it('renders dates and time in the japanese format', () => {
		const originalThemeDisplay = themeDisplay;
		themeDisplay = {getLanguageId: () => 'ja_JP'};

		const {getByText} = render(
			<List {...props} data={['2020-12-20 14:48']} type="date_time" />
		);

		expect(getByText('2020/12/20 14:48')).toBeTruthy();

		themeDisplay = originalThemeDisplay;
	});
});
