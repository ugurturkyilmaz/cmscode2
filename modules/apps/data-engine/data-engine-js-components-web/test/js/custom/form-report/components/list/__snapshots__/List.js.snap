// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders 1`] = `
<DocumentFragment>
  <div
    class="field-list"
  >
    <ul
      class="entries-list"
    >
      <li>
        label1
      </li>
      <li>
        label2
      </li>
      <li>
        label3
      </li>
      <li>
        label4
      </li>
      <li>
        label5
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
